### PR TITLE
Split cosmetic matching from injection

### PR DESCRIPTION
**Pull request by chrmod**
For extended control over cosmetic filters injection we offer an option to match filters and inject them in separate steps. This provides users an option to control which filters should be injected or provide alternative injection method.

### CODE REVIEW

**Comment by seia-soto** [comment]
@chrmod Can we add deprecation notice for `getCosmeticFilters` method while keeping it as-is? I don't think we need to keep that method in long term.

File: packages/adblocker/src/engine/engine.ts [hunk 1]
```diff
@@ -1024,6 +1024,169 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
...
+
+    return {
+      active: true,
+      extended,
+      scripts,
+      styles,
+    };
+  }
+
+  public injectCosmeticFilters(
```
Comment (seia-soto) [inline packages/adblocker/src/engine/engine.ts:1068]: This function doesn't actually inject something to DOM or any other contexts. The name is confusing here, how about: `compileCosmeticsFilters`?

File: packages/adblocker/src/engine/engine.ts [hunk 2]
```diff
@@ -1024,6 +1024,169 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
...
+      styles += `\n\n${createStylesheet([filter.getSelector()], hidingStyle)}`;
+    }
+
+    for (const script of scripts) {
+      this.emit('script-injected', script, url);
+    }
+
+    if (styles.length !== 0) {
+      this.emit('style-injected', styles, url);
+    }
```
Comment (seia-soto) [inline packages/adblocker/src/engine/engine.ts:1140]: I think these were meant to be same as `filter-matched` in previous. Wouldn't it be more clear to move these to `matchCosmeticFilters`?

Comment (chrmod) [reply packages/adblocker/src/engine/engine.ts:1140]: those logs require "compiled" stylesheets and scripts, so cannot be emitted on match as those representations does not exist yet

File: packages/adblocker/src/engine/engine.ts (hunk 1 above)
Comment (chrmod) [reply packages/adblocker/src/engine/engine.ts:1068]: adblocker by itself does not inject anything as it is platform agnostic - same thing with the network layer.

I've chosen `inject` as a name as it build on previous nomenclature of the emitted events `style-injected` and `script-injected`. But I don't a strong opinion. 

Alternative names can also be prepare or assemble? @philipp-classen @remusao @seia-soto  what do you think?

**Comment by chrmod** [comment]
> @chrmod Can we add deprecation notice for `getCosmeticFilters` method while keeping it as-is? I don't think we need to keep that method in long term.

How would you like to communicate the deprecation? But, removing the method will be a breaking change so will require a major version bump.

**Review by philipp-classen — APPROVED** [review]

**Pull request merged**